sentry-android-core/src/main/java/io/sentry/android/core/AnrIntegration.java

```diff
@@ -20,10 +22,17 @@
  */
 public final class AnrIntegration implements Integration, Closeable {
 
+  private final @NotNull Context context;
+
+  public AnrIntegration(final @NotNull Context context) {
+    this.context = context;
+  }
+
   /**
    * Responsible for watching UI thread. being static to avoid multiple instances running at the
    * same time.
    */
+  @SuppressLint("StaticFieldLeak") // we have watchDogLock to avoid those leaks
```

Comment:
we can make it non-final, but then its hard to guarantee that this integration won't be registered twice.

Reply:
We create an instance under a lock and set the instance to null on close so looks good already